addons/addon-base-raas/packages/base-raas-services/lib/plugins/env-provisioning-plugin.js

```diff
@@ -118,11 +118,15 @@ async function updateEnvOnProvisioningSuccess({
   const existingEnvRecord = await environmentScService.mustFind(requestContext, { id: envId, fields: ['rev'] });
 
   // Create DNS record for RStudio workspaces
-  const connectionType = _.find(outputs, o => o.OutputKey === 'MetaConnection1Type').OutputValue;
-  if (connectionType.toLowerCase() === 'rstudio') {
-    const dnsName = _.find(outputs, o => o.OutputKey === 'Ec2WorkspaceDnsName').OutputValue;
-    const environmentDnsService = await container.find('environmentDnsService');
-    environmentDnsService.createRecord('rstudio', envId, dnsName);
+  const connectionType = _.find(outputs, o => o.OutputKey === 'MetaConnection1Type');
```

Comment:
Looks like these two blocks are exactly the same except for the last line, maybe we can extract it to a function and pass in 'delete' or 'create' as a variable?

Reply:
I didn't choose to do that since all the other methods in the plugin are linked to workflow steps. Creating a "private" method in this plugin seemed like an anti-pattern.

Reply:
Also, I'm not sure if we'll need special logic in updateEnvOnTerminationSuccess() since RStudio will soon have auto-stop enabled, and the changed DNS may not be found in the outputs anymore.

Reply:
Got it. Yea, that makes sense. Thanks for explaining this 👍

Reply:
I think a private method can be created in order to avoid duplicate code and it shouldn't be treated as an anti-pattern even though other public methods are linked to workflow steps. The decision to not move the code for determining `dnsName` due to `updateEnvOnTerminationSuccess` changing later is entirely up to you though.